lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -65,111 +66,118 @@ inline bool varianceComparison(const std::pair<UnsignedInteger, Scalar> &a, cons
+  for (UnsignedInteger i = 0; i < outputDimension; ++i)
+    stdDev[i] = std::sqrt(covarianceMatrix(i, i));
 
   // quick summary
-  oss << " input dimension: " << inputDimension << "\n"
-      << " output dimension: " << outputDimension << "\n"
-      << " basis size: " << functionalChaosResult_.getReducedBasis().getSize() << "\n"
-      << " mean: " << mean.__str__() << "\n"
-      << " std-dev: " << stdDev.__str__() << "\n";
-  oss << String(60, '-') << "\n";
-  String st;
-
-  EnumerateFunction enumerateFunction(functionalChaosResult_.getOrthogonalBasis().getEnumerateFunction());
+  oss << "- input dimension=" << inputDimension << "\n"
+      << "- output dimension=" << outputDimension << "\n"
+      << "- basis size=" << functionalChaosResult_.getReducedBasis().getSize() << "\n"
+      << "- mean=" << mean.__str__() << "\n"
+      << "- std-dev=" << stdDev.__str__() << "\n";
+  oss << "\n";
+  String intermediateString;
+  const Scalar varianceThreshold = ResourceMap::GetAsScalar("FunctionalChaosSobolIndices-VariancePartThreshold");
+  const UnsignedInteger maximumNumberOfOutput = ResourceMap::GetAsUnsignedInteger("FunctionalChaosSobolIndices-MaximumNumberOfOutput");
+  const UnsignedInteger columnWidth = ResourceMap::GetAsUnsignedInteger("FunctionalChaosSobolIndices-PrintColumnWidth");
 
   for (UnsignedInteger m = 0; m < outputDimension; ++ m)
   {
-    UnsignedInteger maxdegree = 0;
-
-    std::vector< std::pair<UnsignedInteger, Scalar > > varianceOrder;
-
-    // compute part of contribution of each basis term
-    for (UnsignedInteger i = 1; i < basisSize; ++ i)
-    {
-      Scalar coefI = coefficients(i, m);
-      Indices multiIndices(enumerateFunction(indices[i]));
-      UnsignedInteger degreeI = 0;
-      for (UnsignedInteger k = 0; k < multiIndices.getSize(); ++ k) degreeI += multiIndices[k];
-
-      maxdegree = std::max(maxdegree, degreeI);
-      const Scalar varianceRatio = coefI * coefI / variance[m];
-      varianceOrder.push_back(std::pair<UnsignedInteger, Scalar >(i, varianceRatio));
-    }
-
-    // sort basis terms by descending variance contribution
-    std::sort(varianceOrder.begin(), varianceOrder.end(), varianceComparison);
-
+    if (m > maximumNumberOfOutput) break;
     if (outputDimension > 1)
       oss << "Marginal: " << m << "\n";
 
+    const Point partOfVariance(getPartOfVariance(m));
+    const Sample partOfVarianceSample(Sample::BuildFromPoint(partOfVariance));
+    const Indices order(partOfVarianceSample.argsort(false));
+
     // table of part of variance for each basis term
-    oss << "Index   | Multi-indice                  | Part of variance  \n";
-    oss << String(60, '-') << "\n";
-    for (UnsignedInteger i = 0; i < varianceOrder.size(); ++ i)
+    // Print header
+    oss << "| Index |"
+        << padString(" Multi-index") << "|"
+        << padString(" Part of variance") << "|"
+        << "\n";
+    // Print dashes
+    oss << "|-------|";
+    oss << String(columnWidth, '-') << "|"
```

Comment:
Create a `columnDashes` string with `String(columnWidth, '-')` and use it everywhere appropriate.